src/styles/createMuiTheme.d.ts

```diff
@@ -24,6 +24,9 @@ export type Theme<T = {}> = {
   transitions: Transitions;
   spacing: Spacing;
   zIndex: ZIndex;
+  overrides?: {
+    [name: string]: {}
+  };
```

Comment:
This should probably be

```ts
overrides?: StyleRules;
```

Reply:
Or like this?
```
overrides?: {
   [name: string]: StyleRules
}
```

In styles\getStylesCreator.js I can see 
```
const overrides = theme.overrides[name]
```

And in https://material-ui-next.com/customization/themes/ page, there is **Customizing all instances of a component type** example. I thing that ```name``` is component name, defined in ```withStyles(styles, { name: XXX })```.